tracks/views.py

```diff
@@ -21,10 +21,12 @@ def get(self, request, genome_id):
                 category_obj = Category.objects.get(id=track.category_id)
                 categories[track.category_id] = CategorySerializer(category_obj).data
                 categories[track.category_id]["track_list"] = []
-            categories[track.category_id]["track_list"].append(ReadTrackSerializer(track).data)
-        return Response([categories[category_id] for category_id in categories], status=status.HTTP_200_OK)
+            categories[track.category_id]["track_list"].append(CategoryTrackSerializer(track).data)
+        return Response({"track_categories": [categories[category_id] for category_id in categories]}, status=status.HTTP_200_OK)
     
     def delete(self, request, genome_id):
+        if(settings.DEPLOYMENT_ENV not in ["local","dev","internal","staging"]):
```

Comment:
Why not use different settings file for different environment ? Which is why it is there at the first place. 
I think http_method_names can be utilised here to restrict which methods are allowed which can be set through different settings file.
https://docs.djangoproject.com/en/4.2/ref/class-based-views/base/ and https://docs.djangoproject.com/en/4.2/ref/class-based-views/base/#django.views.generic.base.View.http_method_names

Reply:
I agree, but I could not find better alternatives. Can you explain how do I set the view `http_method_names` through a settings file? 
DEPLOYMENT_ENV is a Django settings variable which is set by secret/configmap file in k8s deployments.

Reply:
`DEPLYOMENT_ENV` is already using a different settings file (set by secret/configmap in k8s deployments) but yes, it adds a filter list in code. I'll try replacing it with environment-sourced `http_method_names`

Reply:
Done.